config/config.go

```diff
@@ -166,14 +223,327 @@ func (cs configs) ResolveConfig(f func(configs []interface{}) error) error {
 // The default configuration sources are:
 // * Environment Variables
 // * Shared Configuration and Shared Credentials files.
-func LoadDefaultConfig(cfgs ...Config) (aws.Config, error) {
-	var cfgCpy configs
-	cfgCpy = append(cfgCpy, cfgs...)
+func LoadDefaultConfig(ctx context.Context, optFns ...func(*LoadOptions) error) (aws.Config, error) {
+	var cfg aws.Config
+	options := LoadOptions{}
+
```

Comment:
styling nit, but prefer `var options LoadOptions` similar style for API clients, and `aws.Config` above. 

Could also do named returns for `cfg` instead of var here but no preference either way.